go/glome/glome.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package glome implements GLOME protocol.
+package glome
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"golang.org/x/crypto/curve25519"
+	"io"
+)
+
+const (
+	// PrivateKeySize is the size of a PrivateKey in bytes.
+	PrivateKeySize = 32
+	// PublicKeySize is the size of a PublicKey in bytes.
+	PublicKeySize = 32
```

Comment:
Could it be that private and public keys have different sizes?

Reply:
In this implementation with the standard GLOME protocol variation is not possible. In other protocols (or in other implementations) it is possible that public/private keys have different values (e.g. https://github.com/golang/go/blob/master/src/crypto/ed25519/ed25519.go)